packages/wdio-config/src/lib/ConfigParser.ts

```diff
@@ -276,15 +276,26 @@ export default class ConfigParser {
         const fileList = ConfigParser.getFilePaths(config, undefined, this._pathService)
         cliArgFileList.forEach(filteredFile => {
             filteredFile = removeLineNumbers(filteredFile)
-            let globMatchedFiles = ConfigParser.getFilePaths(this._pathService.glob(filteredFile), undefined, this._pathService)
+            // Send single file/file glob to getFilePaths - not supporting hierarchy in spec
+            // Return value will alwyas be string[]
+            let globMatchedFiles = <string[]> ConfigParser.getFilePaths(this._pathService.glob(filteredFile), undefined, this._pathService)
             if (this._pathService.isFile(filteredFile)) {
                 filesToFilter.add(this._pathService.ensureAbsolutePath(filteredFile))
             } else if (globMatchedFiles.length) {
                 globMatchedFiles.forEach(file => filesToFilter.add(file))
             } else {
+                // fileList can be a string[] or a string[][]
                 fileList.forEach(file => {
-                    if (file.match(filteredFile)) {
-                        filesToFilter.add(file)
+                    if (typeof file === 'string') {
```

Comment:
files that do not match are silently ignored

Reply:
Added a log message